adapter/internal/ingress/client_manager.go

```diff
@@ -23,9 +23,10 @@ type connection struct {
 type ClientManager struct {
 	connector     ConnectionBuilder
 	connectionTTL time.Duration
-	connections   []unsafe.Pointer
+	connections   []*connection
 	nextIdx       uint64
 	retryWait     time.Duration
+	mu            sync.RWMutex
```

Comment:
Here is where I picked up that convention, in case you were questioning if I fabricated that out of thin air: https://www.youtube.com/watch?v=1rZ-JorHJEY&t=7m57s